### PR TITLE
QF-1001 Uninstall NewtonSoft.Json and install System.Text.Json

### PR DESCRIPTION
**Pull request by SarahDuncan**
Migrate from NewtonSoft.Json to System.Text.Json.

### CODE REVIEW

File: src/SFA.DAS.QnA.Application/Commands/SetPageAnswers/SetAnswersBase.cs [hunk 1]
```diff
@@ -55,7 +55,7 @@ protected void UpdateApplicationData(string pageId, List<Answer> submittedAnswer
         {
             if (application != null)
             {
-                var applicationData = JObject.Parse(application.ApplicationData ?? "{}");
+                var applicationData = JsonObject.Parse(application.ApplicationData ?? "{}");
```
Comment (SawNawDfE) [inline src/SFA.DAS.QnA.Application/Commands/SetPageAnswers/SetAnswersBase.cs:58]: This returns a `JsonNode` (not a `JsonObject`), but since we always cast it to a `JsonObject` before using it, might as well do a one-time cast here instead and remove the two casts down below

```suggestion
                var applicationData = JsonNode.Parse(application.ApplicationData ?? "{}").AsObject();
```

**Review by SawNawDfE — CHANGES_REQUESTED** [review]

File: src/SFA.DAS.QnA.Application/Queries/Sections/GetSequenceSections/GetSequenceSectionsHandler.cs [hunk 2]
```diff
@@ -47,7 +48,7 @@ public async Task<HandlerResponse<List<Section>>> Handle(GetSequenceSectionsRequ
         
         private  void RemovePages(Data.Entities.Application application, Section section)
```
Comment (SawNawDfE) [inline src/SFA.DAS.QnA.Application/Queries/Sections/GetSequenceSections/GetSequenceSectionsHandler.cs:49]: Extra whitespace

File: src/SFA.DAS.QnA.Application/Queries/Sections/GetSequenceSections/GetSequenceSectionsHandler.cs [hunk 3]
```diff
@@ -58,7 +59,7 @@ private static void RemoveInactivePages(Section section)
             section.QnAData.Pages.RemoveAll(p => !p.Active);
         }
 
-        private  void RemovePagesBasedOnNotRequiredConditions(Section section, JObject applicationData)
+        private  void RemovePagesBasedOnNotRequiredConditions(Section section, JsonObject applicationData)
```
Comment (SawNawDfE) [inline src/SFA.DAS.QnA.Application/Queries/Sections/GetSequenceSections/GetSequenceSectionsHandler.cs:62]: extra whitespace

File: src/SFA.DAS.QnA.Application/Commands/Files/UploadFile/SubmitPageOfFilesHandler.cs [hunk 4]
```diff
@@ -161,7 +160,7 @@ private void UpdateApplicationData(SubmitPageOfFilesRequest request, Data.Entiti
         {
             if (application != null)
             {
-                var applicationData = JObject.Parse(application.ApplicationData ?? "{}");
+                var applicationData = JsonObject.Parse(application.ApplicationData ?? "{}");
```
Comment (SawNawDfE) [inline src/SFA.DAS.QnA.Application/Commands/Files/UploadFile/SubmitPageOfFilesHandler.cs:163]: Same thing here, we can cast this to a `JsonObject` instead of doing it twice below

**Review by ChrisJWoodcock — CHANGES_REQUESTED** [review]
If I search for NewtonSoft.Json in the branch QF-1001 I still get lots of hit - other than the schema validation which was discussed - if you could check and say why that is?

File: src/SFA.DAS.QnA.Application/SFA.DAS.QnA.Application.csproj [hunk 5]
```diff
@@ -4,15 +4,21 @@
         <TargetFramework>net6.0</TargetFramework>
     </PropertyGroup>
 
+    <ItemGroup>
+      <Compile Remove="Commands\StartApplication\Helpers\**" />
+      <EmbeddedResource Remove="Commands\StartApplication\Helpers\**" />
+      <None Remove="Commands\StartApplication\Helpers\**" />
+    </ItemGroup>
+
```
Comment (SawNawDfE) [inline src/SFA.DAS.QnA.Application/SFA.DAS.QnA.Application.csproj:12]: Why were these added?

**Review by SawNawDfE — CHANGES_REQUESTED** [review]

File: src/SFA.DAS.QnA.Application/Commands/SetPageAnswers/SetAnswersBase.cs (hunk 1 above)
Comment (SarahDuncan) [reply src/SFA.DAS.QnA.Application/Commands/SetPageAnswers/SetAnswersBase.cs:57]: done

File: src/SFA.DAS.QnA.Application/Queries/Sections/GetSequenceSections/GetSequenceSectionsHandler.cs (hunk 2 above)
Comment (SarahDuncan) [reply src/SFA.DAS.QnA.Application/Queries/Sections/GetSequenceSections/GetSequenceSectionsHandler.cs:48]: not visible in visual studio

File: src/SFA.DAS.QnA.Application/Queries/Sections/GetSequenceSections/GetSequenceSectionsHandler.cs (hunk 3 above)
Comment (SarahDuncan) [reply src/SFA.DAS.QnA.Application/Queries/Sections/GetSequenceSections/GetSequenceSectionsHandler.cs]: not visible in visual studio

File: src/SFA.DAS.QnA.Application/SFA.DAS.QnA.Application.csproj (hunk 5 above)
Comment (SarahDuncan) [reply src/SFA.DAS.QnA.Application/SFA.DAS.QnA.Application.csproj]: removed

File: src/SFA.DAS.QnA.Application/Queries/Sections/GetSequenceSections/GetSequenceSectionsHandler.cs (hunk 2 above)
Comment (SawNawDfE) [reply src/SFA.DAS.QnA.Application/Queries/Sections/GetSequenceSections/GetSequenceSectionsHandler.cs:48]: It is visible for me:

image

Also visible in Visual Studio Code

File: src/SFA.DAS.QnA.Application/Queries/Sections/GetSequenceSections/GetSequenceSectionsHandler.cs (hunk 3 above)
Comment (SawNawDfE) [reply src/SFA.DAS.QnA.Application/Queries/Sections/GetSequenceSections/GetSequenceSectionsHandler.cs]: image

Comment (SarahDuncan) [reply src/SFA.DAS.QnA.Application/Queries/Sections/GetSequenceSections/GetSequenceSectionsHandler.cs]: done

File: src/SFA.DAS.QnA.Application/Queries/Sections/GetSequenceSections/GetSequenceSectionsHandler.cs (hunk 2 above)
Comment (SarahDuncan) [reply src/SFA.DAS.QnA.Application/Queries/Sections/GetSequenceSections/GetSequenceSectionsHandler.cs]: done

**Comment by SawNawDfE** [comment]
I've set this as Draft since we have a bit of work to do on this

**Review by SawNawDfE — CHANGES_REQUESTED** [review]

File: src/SFA.DAS.QnA.Api.Views/Views/QuestionPartials/_TabularData.cshtml [hunk 6]
```diff
@@ -1,10 +1,9 @@
-@using Newtonsoft.Json
 @model dynamic
 
 @{
     var isError = Model.ErrorMessages != null && Model.ErrorMessages.Count > 0 ? "govuk-input--error" : "";
 
-    var tableData = JsonConvert.DeserializeObject<dynamic>(Model.Value);
+    var tableData = JsonSerializer.DeSerialize<dynamic>(Model.Value);
```
[the remark this answers is not in the document]
Comment (SarahDuncan) [reply src/SFA.DAS.QnA.Api.Views/Views/QuestionPartials/_TabularData.cshtml:6]: done
